pkg/executor/sortexec/parallel_sort_worker.go

```diff
@@ -73,9 +72,9 @@ func newParallelSortWorker(
 		timesOfRowCompare:      0,
 		memTracker:             memTracker,
 		sortedRowsIter:         sortedRowsIter,
-		maxSortedRowsLimit:     maxSortedRowsLimit,
+		maxSortedRowsLimit:     maxChunkSize * 30,
 		spillHelper:            spillHelper,
-		batchRows:              make([]chunk.Row, 0, maxSortedRowsLimit),
+		batchRows:              make([]chunk.Row, 0),
```

Comment:
You should set a small init size unless slices have to resize.

Reply:
> You should set a small init size unless slices have to resize.

We don't need to reserve small init size any more as I save chunks and pre-allocate whole memory before sort.